examples/zigzag/main.go

```diff
@@ -0,0 +1,245 @@
+// Copyright 2020 The Ebiten Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// +build example jsgo
+
+package main
+
+import (
+	"fmt"
+	"image/color"
+	"log"
+	"math/rand"
+
+	"github.com/hajimehoshi/ebiten"
+	"github.com/hajimehoshi/ebiten/ebitenutil"
+	"github.com/hajimehoshi/ebiten/inpututil"
+)
+
+const (
+	screenWidth  = 640
+	screenHeight = 480
+	Bit32Max     = 0xFFFFFFFF
+)
+
+var (
+	emptyImage *ebiten.Image
+)
+
+// Scale ...
+type Scale struct {
+	X int64
+	Y int64
+}
+
+// Game ...
+type Game struct {
+	moveDirection int8
+	AppleX        int64
+	AppleY        int64
+	scalesLoc     map[int64]*Scale
+	timer         uint64
+	speed         int8
+	Score         int64
+	bestScore     int64
+	Level         int8
+	nearapple     bool
+}
+
+func init() {
+	emptyImage, _ = ebiten.NewImage(1, 1, ebiten.FilterDefault)
+	_ = emptyImage.Fill(color.White)
+}
+
+func (g *Game) collides() bool {
+	return g.scalesLoc[0].X < g.AppleX+10 &&
+		g.scalesLoc[0].X+10 > g.AppleX &&
+		g.scalesLoc[0].Y < g.AppleY+10 &&
+		g.scalesLoc[0].Y+10 > g.AppleY
+}
+
+func (g *Game) nearApple() bool {
+	return g.scalesLoc[0].X < (g.AppleX-30)+60 &&
+		g.scalesLoc[0].X+10 > g.AppleX-30 &&
+		g.scalesLoc[0].Y < (g.AppleY-30)+50 &&
+		g.scalesLoc[0].Y+10 > g.AppleY-30
+}
+
+func (g *Game) collidesWithSelf() bool {
+	for i, v := range g.scalesLoc {
+		if i > 0 {
+			if g.scalesLoc[0].X < v.X+10 &&
+				g.scalesLoc[0].X+10 > v.X &&
+				g.scalesLoc[0].Y < v.Y+10 &&
+				g.scalesLoc[0].Y+10 > v.Y {
+				return true
+			}
+		}
+	}
```

Comment:
```
	for i, v := range g.scalesLoc[1:] {
		if g.scalesLoc[0].X < v.X+10 &&
			g.scalesLoc[0].X+10 > v.X &&
			g.scalesLoc[0].Y < v.Y+10 &&
			g.scalesLoc[0].Y+10 > v.Y {
			return true
		}
	}
	return false
```

Reply:
Since, I am using scalesLoc as a map. Go errors, not letting me slice index.

Reply:
Acknowledged.